toolchain/check/handle_function.cpp

```diff
@@ -263,6 +264,17 @@ static auto BuildFunctionDecl(Context& context,
                           name, decl_id, is_extern, introducer.extern_library)},
                       {.return_slot_pattern_id = name.return_slot_pattern_id,
                        .virtual_modifier = virtual_modifier}};
+  // Find self parameter pattern.
+  // TODO: Do this during initial traversal of implicit params.
+  for (auto implicit_param_id : context.inst_blocks().GetOrEmpty(
+           function_info.implicit_param_patterns_id)) {
+    if (SemIR::Function::GetNameFromPatternId(
+            context.sem_ir(), implicit_param_id) == SemIR::NameId::SelfValue) {
+      CARBON_CHECK(!function_info.self_param_id.has_value());
+      function_info.self_param_id = implicit_param_id;
```

Comment:
nit: This could set a local variable before the Function construction, and then we could unconditionaly set .self_param_id during construction, in the aggregate initializer, instead of mutating `function_info` afterward?

Reply:
0d33379db7fc8a4bc74960e04a9e2413fbce1ac0